src/plugins/intel_gpu/tests/unit/test_cases/gemm_gpu_test.cpp

```diff
@@ -1739,6 +1739,52 @@ TEST(gemm_onednn, impl_replacement_with_cldnn) {
     ASSERT_FALSE(impl->is_dynamic());
 }
 
+TEST(gemm_onednn, input0_simple_input1_fsv16_select_ref_kernel) {
+    auto& engine = get_test_engine();
+    if (!engine.get_device_info().supports_immad)
+        return;
+
+    tests::random_generator rg(GET_SUITE_NAME);
+    int input_b = 1, input_f = 16, input_y = 3, input_x = 3;
+    int output_f = 16;
+
+    auto input_size = tensor(input_b, input_f, input_x, input_y);
+    auto input_data = rg.generate_random_4d<ov::float16>(input_b, input_f, input_y, input_x, -1, 1);
+    auto input_data_bfyx = flatten_4d(format::bfyx, input_data);
+    auto input_mem = engine.allocate_memory({ data_types::f16, format::bfyx, input_size });
+    set_values(input_mem, input_data_bfyx);
+
+    auto weights_size = tensor(16, 16, 1, 1, 1);
+    auto weights_data = rg.generate_random_4d<ov::float16>(output_f, input_f, 1, 1, -1, 1);
+    auto weights_data_bfyx = flatten_4d(format::bfyx, weights_data);
+    auto weights_mem = engine.allocate_memory({ data_types::f16, format::bfyx, weights_size });
+    set_values(weights_mem, weights_data_bfyx);
+
+    topology topology_test(input_layout("input", input_mem->get_layout()),
+        data("weights", weights_mem),
+        reorder("input_fsv", input_info("input"), { data_types::f16, format::b_fs_yx_fsv16, input_size }),
+        convolution("conv1", input_info("input_fsv"), "weights", "", 1, {1, 1}, {1, 1}, {0, 0}, {0, 0}, false),
+        convolution("conv2", input_info("conv1"), "weights", "", 1, { 1, 1 }, { 1, 1 }, { 1, 1 }, { 2, 2 }, false),
```

Comment:
I'm not sure what you want to fix whether fixing opt pass relating part or just fixing this unittest like removing conv2.
This unittest is designed to reproduce target model's issue such a pattern conv(onednn)-conv-gemm(ocl). So if the pattern is different, it would not happen target issue.

Reply:
This test case looks weird because it depends on unexpected behavior of conv selection. ref kernel / ocl kernel is not supposed to be chosen from dGPU. I'm afraid this might cause redundant work in the future. Because of that, I'd like to suggest to remove this test case and just include the code only.

Reply:
Deleted this testcase.